snd-server/recommendation_engine.py

```diff
@@ -0,0 +1,100 @@
+import joblib
+import zipfile
+import pandas as pd
+
+from typing import List
+from sklearn.metrics.pairwise import cosine_similarity
+
+from models.track import Track, Album, Artist, RecommendedTrack
+from context import Context
+from app import create_ctx
+
+
+class Recommender:
+    def __init__(self, ctx: Context) -> None:
+        self.config = ctx.config
+        self.zf = zipfile.ZipFile(self.config.archive_storage_path, mode="r")
+        self.scaler = joblib.load(self.zf.open(self.config.scaler_storage_path))
+        self.dataset = pd.read_csv(
+            self.zf.open(self.config.transformed_sounds_storage_path), index_col=0
+        )
+
+    def __prepare_track(self, track: Track):
+        prepared_track = pd.Series(track.dict())
+        prepared_track = prepared_track.drop(labels=["id", "name", "album", "artists"])
+
+        prepared_track = self.scaler.transform(prepared_track.values.reshape(1, -1))
+
+        return prepared_track[0]
+
+    def get_top_n(self, track: Track, n: int) -> List[RecommendedTrack]:
+        prepared_track = self.__prepare_track(track)
+        print(self.dataset)
```

Comment:
This will probably flood the logs too much, perhaps try to log more specific values that are interesting to inspect.
Also you can use `logger = logging.getLogger(__name__)` [instead of print](https://github.com/spris-dev/soundations/blob/master/snd-server/services/sqlite_storage.py#L28)